pkg/controller/seed-controller-manager/default-application-controller/controller.go

```diff
@@ -156,6 +159,24 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluste
 		ignoreDefaultApplications = true
 	}
 
+	userClusterClient, err := r.userClusterConnectionProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user cluster client: %w", err)
+	}
+
+	if cluster.Spec.CNIPlugin != nil && cni.IsManagedByAppInfra(cluster.Spec.CNIPlugin.Type, cluster.Spec.CNIPlugin.Version) {
+		ciliumApp, err := util.GetCNIApplicationInstallation(ctx, userClusterClient, cluster.Spec.CNIPlugin.Type)
+		if err != nil {
+			r.log.Debug("Requeue as it could not get Cilium system ApplicationInstallation")
+			return &reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		}
+		// check if application is deployed and status is updated with app version.
+		if ciliumApp != nil && ciliumApp.Status.ApplicationVersion == nil {
+			r.log.Debug("Requeue as Cilium system application is not ready yet")
+			return &reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Even if we use the same code somewhere else already, I think we should add a test for this as well in here. We do the same in the initial app-controller logic.
https://github.com/kubermatic/kubermatic/blob/0033300c919b3b35da913a23eabb6b394cf49e8e/pkg/controller/seed-controller-manager/initial-application-installation-controller/controller_test.go#L165